src/wmtk/Mesh.hpp

```diff
@@ -112,6 +114,31 @@ class Mesh
     Accessor<char> get_flag_accessor(PrimitiveType type);
     Accessor<long> get_cell_hash_accessor();
 
+    /**
+     * @brief update hashes in given cells
+     *
+     * @param cells vector of tuples in which the hash should be updated
+     * @param hash_accessor hash accessor
+     */
+    void update_cell_hashes(const std::vector<Tuple>& cells, Accessor<long>& hash_accessor);
+    /**
+     * @brief same as `update_cell_hashes` but slow because it creates a new accessor
+     */
+    void update_cell_hashes_slow(const std::vector<Tuple>& cells);
+
+    /**
+     * @brief return the same tuple but with updated hash
+     *
+     * @param tuple tuple with potentially outdated hash
+     * @param hash_accessor hash accessor
+     * @return tuple with updated hash
+     */
+    Tuple resurrect_tuple(const Tuple& tuple, Accessor<long>& hash_accessor) const;
```

Comment:
this will need to be a constaccessor. just copypaste two versions for now and make an issue (assigned to me) to fix accessor inheritance so accessors can be implicitly converted to constaccessors

Reply:
oh and mark the non-const accessor one as deprecated just so we know to delete it ASAP?